### PR TITLE
coprocessor: add error to coprocessor

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang PTAL

### CODE REVIEW

File: src/raftstore/coprocessor/dispatcher.rs [hunk 1]
```diff
@@ -143,19 +148,22 @@ mod test {
...
+        return_err: Arc<RwLock<bool>>,
     }
 
     impl TestCoprocessor {
         fn new(bypass_pre: Arc<RwLock<bool>>,
                bypass_post: Arc<RwLock<bool>>,
                called_pre: Arc<RwLock<u8>>,
-               called_post: Arc<RwLock<u8>>)
+               called_post: Arc<RwLock<u8>>,
+               return_err: Arc<RwLock<bool>>)
```
Comment (ngaut) [inline src/raftstore/coprocessor/dispatcher.rs]: Can we remove this argument? I guess it should be enabled by default.

Comment (BusyJay) [inline src/raftstore/coprocessor/dispatcher.rs]: No, we use this argument to emulate a condition that return a error in the middle running.

Comment (ngaut) [inline src/raftstore/coprocessor/dispatcher.rs]: Then we should comment it "For testing only"

Comment (ngaut) [inline src/raftstore/coprocessor/dispatcher.rs]: Ah, implementation and tests are in the same file.

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/coprocessor/error.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::result::Result as StdResult;
+use std::error::Error as StdError;
+
+
+quick_error!{
+    #[derive(Debug)]
+    pub enum Error {
```
Comment (siddontang) [inline src/raftstore/coprocessor/error.rs]: why not using raftstore error directly?

Comment (BusyJay) [inline src/raftstore/coprocessor/error.rs]: It would be more clear when we log the error.

Comment (siddontang) [inline src/raftstore/coprocessor/error.rs]: how to more clear?

**Comment by siddontang** [comment]
LGTM

**Pull request merged**